### PR TITLE
Add O_SEARCH to all platforms supported by libc crate

### PR DESCRIPTION
**Pull request by TheJonny**
## Checklist:

- [x] I have read `CONTRIBUTING.md`
- [x] I have written necessary tests and rustdoc comments
- [ ] A change log has been added if this PR modifies nix's API

### CODE REVIEW

File: src/fcntl.rs [hunk 1]
```diff
@@ -164,8 +164,9 @@ libc_bitflags!(
         /// Similar to `O_DSYNC` but applies to `read`s instead.
         #[cfg(any(target_os = "linux", netbsdlike))]
         O_RSYNC;
-        /// Skip search permission checks.
-        #[cfg(target_os = "netbsd")]
+        /// Open directory for search only. Skip search permission checks on
+        /// later `openat()` calls using the obtained file descriptor.
+        #[cfg(any(target_os = "netbsd", freebsd, solarish, target_os = "fuchsia"))]
```
Comment (SteveLauC) [inline src/fcntl.rs:169]: [We don use alias for only 1 platform](https://github.com/nix-rust/nix/blob/master/CONVENTIONS.md#cfg-gates):

```suggestion
        #[cfg(any(target_os = "netbsd", target_os = "freebsd", solarish, target_os = "fuchsia"))]
```

**Comment by SteveLauC** [comment]
Hi, thanks for contributing to Nix, this patch LGTM expect that alias:)

**Comment by TheJonny** [comment]
Should we also add it in all the other platforms supported by libc that are not in the tier list of nix? that would be `aix`, `wasi`,  `emscripten` ?

**Comment by SteveLauC** [comment]
Hi, would you still want to add this symbol for [these platforms](https://github.com/nix-rust/nix/pull/2374#issuecomment-2058362672) :)

**Comment by TheJonny** [comment]
> Hi, would you still want to add this symbol for [these platforms](https://github.com/nix-rust/nix/pull/2374#issuecomment-2058362672) :)

done - I just was scared by the long cfg line :)

**Comment by TheJonny** [comment]
(I would also like to see it on apple, but it's currently missing from libc - https://github.com/rust-lang/libc/pull/3659 )

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**